### ISSUE TITLE
`MessageFormatter::format` has incompatible type declaration with Guzzle

### ISSUE THREAD

**Issue by chescos** [restcord/restcord#148]
I'm getting this error:

```                                                                     
PHP Fatal error:  Declaration of RestCord\Logging\MessageFormatter::format(Psr\Http\Message\RequestInterface $request, $response = NULL, $error = NULL) must be compatible with GuzzleHttp\MessageFormatter::format(Psr\Http\Message\RequestInterface $request, ?Psr\Http\Message\ResponseInterface $response = NULL, ?Throwable $error = NULL): string in /home/vagrant/code/vendor/restcord/restcord/src/Logging/MessageFormatter.php on line 45
```

**Comment by cryptiklemur** [comment]
Can you please look into this @bumbummen99

**Comment by chescos** [comment]
Restcord currently supports Guzzle 6 and 7, I believe the MessageFormat implementation won't work for both versions because the `format` method changed its declaration with Guzzle 7. The error occurs when using Guzzle 7.

**Comment by bumbummen99** [comment]
> 
> 
> Restcord currently supports Guzzle 6 and 7, I believe the MessageFormat implementation won't work for both versions because the `format` method changed its declaration with Guzzle 7. The error occurs when using Guzzle 7.

Yep, exactly that, we have to re add the type hints / revert daa7bdc563dec1df55905d43fd7ffea3c2028986. I'll create an PR 👍 

Edit: Here is the PR https://github.com/restcord/restcord/pull/150

**Comment by cryptiklemur** [comment]
Give `dev-develop` a try @chescos

**Comment by cryptiklemur** [comment]
PR #150 fixed the original Guzzle 7 signature mismatch. RestCord 0.9.0 now uses Guzzle 8 and has removed the old MessageFormatter implementation entirely.
